include/tarxx.h

```diff
@@ -72,15 +76,53 @@ namespace tarxx {
             //TODO support for utar, star, etc.
         };
 
-        using callback_t = std::function<void(const block_t&)>;
+        using callback_t = std::function<void(const block_t&, size_t size)>;
+
+#ifdef WITH_COMPRESSION
+        enum class compression_mode : unsigned {
+            none,
+#    ifdef WITH_LZ4
+            lz4,
+#    endif
+        };
+
+        tarfile(const std::string& filename,
+                compression_mode compression,
```

Comment:
```suggestion
                compression_mode compression = compression_mode::none,
```
let's assume no compression wanted even if we have compression capabilities compiled in.

Reply:
Won't do as there is a separate c'tor for this here https://github.com/thilo-schmitt/tarxx/pull/2/files#diff-3b0157ec682203ed14348cb1ce277ff0f33482ead367adc9a2d0f9654bde6281R120

Reply:
ah, I see, you default-init `compression_` and therefore you don't need it in the ctor-init-list. well, actually `compression_` shouldn't be there if we build using `WITH_COMPRESSION=OFF`. please surround the delcaration of the `compression_` field with `#ifdef WITH_COMPRESSION` and then add above suggestion.